packages/quantic/check-sfdx-project.js

```diff
@@ -0,0 +1,24 @@
+const pathlib = require('path');
+const fs = require('fs');
+
+function getGrandParentPath(path) {
```

Comment:
We need to move up exactly 2 directory level because the scripts will be executed: project_name/node_modules/quantic.
So to get the project name path we take two steps up.

Reply:
I'm concerned about moving up a fixed number of levels here. There are many tools out there that can install packages from npmjs, and I am not sure if we can assume the directory structure.

Would it be possible to try and detect the `sfdx-project.json` file in a more dynamic way? For instance, maybe we could define the "project directory" as the one that contains the `sfdx-project.json` file AND the `node_modules` directory. So if we find a directory containing the `node_modules` subdirectory, but not the `sfdx-project.json` file, then the project is not compliant.

If we go this route, we'd have to be careful not to introduce a "directory traversal" issue though.

Long story short, we should try with a few package manager tools to see how portable the current implementation is.

Reply:
What do you think of having this function instead of moving up a fixed number of levels:
```
function getProjectPath() {
    return __dirname.split('/node_modules')[0];
}
```
this will satisfy the definition you mentioned of  the "project directory".
and this will address the case where for example quantic would be installed by another package. For example we would still get the correct path to the project when being in the following path:
`/home/project/node_modules/pkg1/node_modules/quantic/pre-install-script`